aptos-move/block-executor/src/executor.rs

```diff
@@ -409,13 +413,109 @@ where
+        // If any of the aggregator v2 identifiers in the resource are part of aggregator_v2_write_set,
+        // then include the resource in the write set.
+        let aggregator_v2_keys = last_input_output.aggregator_v2_keys(txn_idx);
+        let read_set = last_input_output.read_set(txn_idx);
+        // TODO: Gives some error without this `mut`. See if we can avoid `mut` here.
+        if let Some(mut aggregator_v2_keys) = aggregator_v2_keys {
+            if let Some(read_set) = read_set {
+                for (key, data_read) in read_set.as_ref().data_reads.iter() {
+                    if write_set_keys.contains(key) {
+                        continue;
+                    }
+                    // layout is Some(_) if it contains an aggregator
+                    if let DataRead::Versioned(_version, value, Some(layout)) = data_read {
+                        if let Some(value_bytes) = value.bytes() {
+                            match latest_view.replace_identifiers_with_values(value_bytes, layout) {
+                                Ok((patched_bytes, aggregator_v2_keys_in_resource)) => {
+                                    let mut changed_aggregator = false;
+                                    for aggregator_v2_key in aggregator_v2_keys_in_resource {
+                                        if aggregator_v2_keys.any(|key| key == aggregator_v2_key) {
+                                            changed_aggregator = true;
+                                            break;
+                                        }
+                                    }
+                                    if changed_aggregator {
+                                        let mut patched_value = value.as_ref().clone();
+                                        patched_value.set_bytes(patched_bytes);
+                                        patched_resource_write_set.insert(key.clone(), patched_value);
+                                    }
+                                },
+                                Err(_) => unreachable!(
+                                    "Failed to replace identifiers with values in read set"
+                                ),
+                            };
+                        } else {
+                            // TODO: Is this unreachable?
+                            unreachable!("Data read value must exist");
+                        }
+                    }
+                }
+            }
+        }
+
+        // For each aggregator v2 in the event, replace aggregator v2 identifier with value.
+        let events = last_input_output.events(txn_idx);
+        let mut patched_events = vec![];
+        for (event, layout) in events {
+            if let Some(layout) = layout {
+                let (_, _, _, event_data) = event.get_event_data();
+                match latest_view
+                    .replace_identifiers_with_values(&Bytes::from(event_data.to_vec()), &layout)
```

Comment:
event_data.to_vec().into()? or event_data.into()

Reply:
Tried it. into() function is not implemented here.